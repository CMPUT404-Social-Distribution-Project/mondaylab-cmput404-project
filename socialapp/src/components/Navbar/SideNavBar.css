
#sidebar {
    position: fixed;
    width: 4rem;
    height: 100%;
    display: flex;
    flex-direction: column;
    padding-top: 2rem;
}


.nav-pills .nav-link {
    border: 0;
    border-radius: 50rem;
}

body {
    min-height: 100vh;
    min-height: -webkit-fill-available;
    
}

html {
    height: -webkit-fill-available;
}

main {
    display: flex;
    flex-wrap: nowrap;
    height: 100vh;
    height: -webkit-fill-available;
    max-height: 100vh;
    overflow-x: auto;
    overflow-y: hidden;
}

hr {
    color: var(--white-teal);
}
.b-example-divider {
    flex-shrink: 0;
    width: 1.5rem;
    height: 100vh;
    background-color: rgba(0, 0, 0, .1);
    border: solid rgba(0, 0, 0, .15);
    border-width: 1px 0;
    box-shadow: inset 0 .5em 1.5em rgba(0, 0, 0, .1), inset 0 .125em .5em rgba(0, 0, 0, .15);
}

.bi {
    vertical-align: -.125em;
    pointer-events: none;
    fill: currentColor;
}

.nav-flush .nav-link {
    border-radius: 0;
}

.scrollarea {
    overflow-y: auto;
}

.fw-semibold { font-weight: 600; }
.lh-tight { line-height: 1.25; }


.signout {
    color: var(--white-teal);
    margin-top: 0.5rem;
    cursor: pointer;

}

.signout:hover {
    color: var(--orange);
}

/* Create Post Button */

#sidebar .nav-item {
    align-self: center;
}

.create-post-button {
    color: var(--dark-blue);
    background-color: var(--teal);
    height: 50px;
    width: 50px;
    margin: 0.5em 0em;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    border: 0;
    border-radius: 11px;
}

.create-post-button:hover {
    cursor: pointer;
}

.nav-link:hover, .nav-link:focus {
    color: var(--orange);
    cursor: pointer;
}
