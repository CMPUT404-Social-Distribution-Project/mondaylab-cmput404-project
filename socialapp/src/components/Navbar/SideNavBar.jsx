import React, { Component, useContext } from "react";
import { Container, Nav, Navbar } from "react-bootstrap";
import "./SideNavBar.css";
import { FaHome, FaInbox, FaSearch, FaSignOutAlt } from "react-icons/fa";
import { MdEdit } from "react-icons/md";
import { Link } from "react-router-dom";
import AuthContext from "../../context/AuthContext";

export default function Sidebar() {

    const { logoutUser } = useContext(AuthContext);


    return (
        <ul
            id="sidebar"
            class="nav nav-pills nav-flush flex-column mb-auto text-center"
        >
            <li class="nav-item">
                <a
                    href="#"
                    class="nav-link active py-3"
                    aria-current="page"
                    title="Create Post"
                    data-bs-toggle="tooltip"
                    data-bs-placement="right"
                >
                    <MdEdit />
                </a>
            </li>
            <li>
                <Link to="/stream">
                    <a
                        href="#"
                        class="nav-link py-3"
                        title="Stream"
                        data-bs-toggle="tooltip"
                        data-bs-placement="right"
                    >
                        <FaHome />
                    </a>
                </Link>
            </li>
            <li>
                <Link to="/inbox">
                    <a
                        href="#"
                        class="nav-link py-3"
                        title="Inbox"
                        data-bs-toggle="tooltip"
                        data-bs-placement="right"
                    >
                        <FaInbox />
                    </a>
                </Link>
            </li>
            <li>
                <Link to="/explore">
                    <a
                        href="#"
                        class="nav-link py-3"
                        title="Explore"
                        data-bs-toggle="tooltip"
                        data-bs-placement="right"
                    >
                        <FaSearch />
                    </a>
                </Link>
            </li>
            <hr class="solid" />
            <div className="signout">
                <FaSignOutAlt 
                    onClick={() => logoutUser()}
                />
            </div>
            
        </ul>
    );
}


