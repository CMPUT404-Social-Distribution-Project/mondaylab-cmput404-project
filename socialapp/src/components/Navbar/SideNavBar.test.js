import React from "react";
import { render } from '@testing-library/react';
<<<<<<< HEAD
import SideNavBar from './SideNavbar';
=======
import SideNavBar from './SideNavBar';
>>>>>>> 4c4581a2a221bdc6186399569d61b92a5569f7a0
import { BrowserRouter } from "react-router-dom";

describe('Testing Side NavBar:', () => {
    it("renders correctly", () => {
        const tree = render(<BrowserRouter><SideNavBar /></BrowserRouter>);
        expect(tree).toMatchSnapshot();
    });
});