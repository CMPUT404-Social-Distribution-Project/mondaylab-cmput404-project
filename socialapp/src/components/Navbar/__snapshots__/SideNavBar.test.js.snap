// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Side NavBar: renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="nav nav-pills nav-flush flex-column mb-auto text-center"
        id="sidebar"
      >
        <li
          class="side-nav-item"
        >
          <div
            aria-current="page"
            class="create-post-button"
            title="Create Post"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </div>
        </li>
        <li
          class="side-nav-item"
        >
          <div
            class="side-nav-link"
            title="Stream"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              />
            </svg>
          </div>
        </li>
        <li
          class="side-nav-item"
        >
          <div
            class="side-nav-link"
            title="Inbox"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M567.938 243.908L462.25 85.374A48.003 48.003 0 0 0 422.311 64H153.689a48 48 0 0 0-39.938 21.374L8.062 243.908A47.994 47.994 0 0 0 0 270.533V400c0 26.51 21.49 48 48 48h480c26.51 0 48-21.49 48-48V270.533a47.994 47.994 0 0 0-8.062-26.625zM162.252 128h251.497l85.333 128H376l-32 64H232l-32-64H76.918l85.334-128z"
              />
            </svg>
          </div>
        </li>
        <li
          class="side-nav-item"
        >
          <div
            class="side-nav-link"
            title="Explore"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </li>
        <hr
          class="solid"
        />
        <div
          class="signout"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
            />
          </svg>
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="nav nav-pills nav-flush flex-column mb-auto text-center"
      id="sidebar"
    >
      <li
        class="side-nav-item"
      >
        <div
          aria-current="page"
          class="create-post-button"
          title="Create Post"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </div>
      </li>
      <li
        class="side-nav-item"
      >
        <div
          class="side-nav-link"
          title="Stream"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            />
          </svg>
        </div>
      </li>
      <li
        class="side-nav-item"
      >
        <div
          class="side-nav-link"
          title="Inbox"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M567.938 243.908L462.25 85.374A48.003 48.003 0 0 0 422.311 64H153.689a48 48 0 0 0-39.938 21.374L8.062 243.908A47.994 47.994 0 0 0 0 270.533V400c0 26.51 21.49 48 48 48h480c26.51 0 48-21.49 48-48V270.533a47.994 47.994 0 0 0-8.062-26.625zM162.252 128h251.497l85.333 128H376l-32 64H232l-32-64H76.918l85.334-128z"
            />
          </svg>
        </div>
      </li>
      <li
        class="side-nav-item"
      >
        <div
          class="side-nav-link"
          title="Explore"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </div>
      </li>
      <hr
        class="solid"
      />
      <div
        class="signout"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
          />
        </svg>
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
